pkgs/development/python-modules/flufl/bounce.nix

```diff
@@ -1,14 +1,49 @@
-{ buildPythonPackage, fetchPypi, atpublic, zope-interface, nose2 }:
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools
+, atpublic
+, zope-interface
+, pytestCheckHook
+}:
 
 buildPythonPackage rec {
-  pname = "flufl.bounce";
+  pname = "flufl-bounce";
   version = "4.0";
-
-  buildInputs = [ nose2 ];
-  propagatedBuildInputs = [ atpublic zope-interface ];
+  pyproject = true;
 
   src = fetchPypi {
-    inherit pname version;
+    pname = "flufl.bounce";
+    inherit version;
     hash = "sha256-JVBK65duwP5aGc1sQTo0EMtRT9zb3Kn5tdjTQ6hgODE=";
   };
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
+    atpublic
+    zope-interface
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [
+    "flufl.bounce"
+  ];
+
+  pythonNamespaces = [
+    "flufl"
+  ];
```

Comment:
Renames seem fine to me, but I think all these extra lines are a net negative to readability.  It wasn't very information dense before, but now the whole very short package definition doesn't even fit in a screenful for me.

Reply:
> It wasn't very information dense before

This is because it is too legacy to package.

Regarding `pythonNamespaces`, flufl [still has the legacy namespaces](https://gitlab.com/warsaw/flufl.bounce/-/blob/4.0/setup.py?ref_type=tags#L12) which are [no longer recommended](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/#pkg-resources-style-namespace-packages).
We need to set pythonNamespaces to package them correctly.
The namespace package is quite tricky and [has blocked sphinx updates for a long time](https://github.com/NixOS/nixpkgs/pull/259350). 

Regarding `pyproject`, we [should no longer use legacy hooks](https://packaging.python.org/en/latest/discussions/setup-py-deprecated/) like `setuptoolsBuildHook`.
We can build it in a modern way by setting `pyproject = true`.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

Regarding `checkPhase`, [previously, `setup.py test` was executed implicitly](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#how-to-provide-automatic-tests-to-python-packages-automatic-tests), so we only needed to add the necessary dependencies to `checkInputs` (now `nativeCheckInputs` is recommended).
However, as explained in the pyproject section, the command `setup.py test` should no longer be used. And this command is never called if `pyproject = true` is set.
https://github.com/NixOS/nixpkgs/blob/7057cc800bd132fc9b13d6be23dbe788632ef807/pkgs/development/interpreters/python/mk-python-derivation.nix#L277-L282

So we need to explicitly pass `pytestCheckHook`.
`pythonImportsCheck` should also be added to all packages as an almost no-cost smoke test, but it would be especially useful for packages that use namespaces like this.

Regarding `meta`, I don't know about the old days, but it should be recommended to set meta now. (Sorry, I could not find the document.)
I don't think it's a proper package if it doesn't even indicate license.

Reply:
I'm not talking about the new attributes, but about the lists that previously took three lines each now being spread over four.

Reply:
That is the style we use throughout python-modules.

Reply:
That makes sense then, I suppose.